testing_and_setup/compass/manage_regression_suite.py

```diff
@@ -193,19 +196,276 @@ def process_test_clean(test_tag, work_dir):  # {{{
-         '-c',  test_configuration, '-r', test_resolution, '-t', test_test],
+         '-c', test_configuration, '-r', test_resolution, '-t', test_test],
         stdout=dev_null, stderr=dev_null)
 
     print("   -- Cleaned case '{}': -o {} -c {} -r {} -t {}".format(
         test_name, test_core, test_configuration, test_resolution, test_test))
 
     dev_null.close()
+    # }}}
 
-# }}}
+
+def local_parallel_setup_script(
+        work_dir, suite_tag, verbose, args, suite_root, testcase_data, testcase_data_prereq):
+    # {{{
+    try:
+        suite_name = suite_tag.attrib['name']
+    except KeyError:
+        print("ERROR: <regression_suite> tag is missing 'name' attribute.")
+        print('Exiting...')
+        sys.exit(1)
+
+    if not os.path.exists('{}'.format(work_dir)):
+        os.makedirs('{}'.format(work_dir))
+
+    if verbose:
+        # flush existing regression suite output file
+        open(work_dir + '/manage_regression_suite.py.out', 'w').close()
+
+    regression_script_name = '{}/parallel_{}.py'.format(work_dir, suite_name)
+    local_parallel_script = open('{}'.format(regression_script_name), 'w')
+    local_parallel_code = write_regression_local_parallel_top(
+        work_dir, suite_tag, args.nodes)
+
+    setup_suite(
+        suite_root,
+        args.work_dir,
+        args.model_runtime,
+        args.config_file,
+        args.baseline_dir,
+        args.verbose,
+        args)
+    local_parallel_code = write_regression_local_parallel_testcase_data(
+        local_parallel_code, work_dir, testcase_data,  testcase_data_prereq)
+    local_parallel_code = write_regression_local_parallel_bottom(
+        local_parallel_code)
+
+    return local_parallel_script, local_parallel_code
+    # }}}
+
+
+def get_prereq(testcase_data, testcase_data_prereq):
+    # {{{
+    names = [i for i in testcase_data.keys() if i in testcase_data_prereq]
+    pre = {}
+    for i in names:
+        pre[i] = testcase_data[i]
+    return pre
+    # }}}
+
+
+def get_testcase_data(testcase_data, testcase_data_prereq):
+    # {{{
+    names = [i for i in testcase_data.keys() if i not in testcase_data_prereq]
+    testcase_data_values = {}
+    for i in names:
+        testcase_data_values[i] = testcase_data[i]
+    return testcase_data_values
+    # }}}
+
+
+def write_regression_local_parallel_testcase_data(local_parallel_code, work_dir, testcase_data, testcase_data_prereq):
+        # {{{
+    index = 0
+    testcase_data_values = get_testcase_data(  testcase_data , testcase_data_prereq)
+    pre_testcase_data    = get_prereq(testcase_data , testcase_data_prereq)
+
+    for key in pre_testcase_data.keys():
+        local_parallel_code += "locations.append('" + \
+            pre_testcase_data[key]['path'] + "')\n"
+        local_parallel_code += "commands.append(['time' , '-p' , '" + os.getcwd(
+        ) + "/" + pre_testcase_data[key]['path'] + "/' + str(run_file_names[{}])])\n".format(str(index))
```

Comment:
```suggestion
        local_parallel_code += "commands.append(['time' , '-p' , '" + work_dir \
                               + "/" + pre_testcase_data[key]['path'] + "/' + str(run_file_names[{}])])\n".format(str(index))
```
This needs to be the work directory, not the current working directory.  (That is the purpose of the `work_dir` -- that's where we plan to run the test suite.)

Finally, I am not a fan of string addition.  There's a lot of it here and maybe that was true before.  But I think the `.format()` method is a much preferable way to got.  Something to keep in the mind in the future.